sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -838,72 +842,102 @@ private async Task RunAsync(CancellationToken cancellationToken)
-                    .Where(ownership => utcNow.Subtract(ownership.LastModifiedTime.Value) < OwnershipExpiration);
-
-                // Dispose of all previous partition ownership instances and get a whole new dictionary.
-
-                InstanceOwnership = activeOwnership
-                    .Where(ownership => ownership.OwnerIdentifier == Identifier)
-                    .ToDictionary(ownership => ownership.PartitionId);
+                await RenewOwnershipAsync(cancellationToken).ConfigureAwait(false);
 
                 // Some previously owned partitions might have had their ownership expired or might have been stolen, so we need to stop
                 // the processing tasks we don't need anymore.
 
                 await Task.WhenAll(ActivePartitionProcessors.Keys
                     .Except(InstanceOwnership.Keys)
-                    .Select(partitionId => StopPartitionProcessingIfRunningAsync(partitionId, ProcessingStoppedReason.OwnershipLost)))
+                    .Select(partitionId => StopPartitionProcessingIfRunningAsync(partitionId, ProcessingStoppedReason.OwnershipLost, cancellationToken)))
                     .ConfigureAwait(false);
 
                 // Now that we are left with processing tasks that should be running, check their status.  If any has stopped, it
                 // means a failure has happened, so try closing it and starting a new one.  In case we don't have a task that should
-                // exist, create it.  This might happen if the user hasn't updated ActivePartitionProcessors when initializing processing
-                // in the previous cycle.
+                // exist, create it.  This might happen if task creation failed in the last cycle.
 
                 await Task.WhenAll(InstanceOwnership
                     .Select(async kvp =>
                     {
                         if (!ActivePartitionProcessors.TryGetValue(kvp.Key, out var t) || t.Item1.IsCompleted)
                         {
-                            await StopPartitionProcessingIfRunningAsync(kvp.Key, ProcessingStoppedReason.Shutdown).ConfigureAwait(false);
-                            await StartPartitionProcessingAsync(kvp.Key).ConfigureAwait(false);
+                            await StopPartitionProcessingIfRunningAsync(kvp.Key, ProcessingStoppedReason.Shutdown, cancellationToken).ConfigureAwait(false);
+                            await StartPartitionProcessingAsync(kvp.Key, cancellationToken).ConfigureAwait(false);
                         }
                     }))
                     .ConfigureAwait(false);
 
-                // Get a complete list of the partition ids present in the Event Hub.  This should be immutable for the time being, but
-                // it may change in the future.
+                // From the storage service provided by the user, obtain a complete list of ownership, including expired ones.  We may still need
```

Comment:
(extreme) nit:  The comment isn't entirely accurate these days, due to the removal of PartitionManager from the public eye, no?